src/main/java/mate/academy/service/impl/BookServiceImpl.java

```diff
@@ -57,14 +65,27 @@ public void deleteById(Long id) {
 
     @Override
     public List<BookDto> findByParameters(BookSearchParametersDto bookSearchDto,
-                                          Pageable pageable) {
+                                                            Pageable pageable) {
         return bookRepository.findAll(builder.buildSpecification(bookSearchDto), pageable)
                 .stream()
                 .map(bookMapper::toDto)
                 .toList();
     }
 
+    @Override
+    public List<BookDtoWithoutCategoryIds> findAllByCategoryId(Long categoryId) {
+        return bookRepository.findByCategories_Id(categoryId).stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .toList();
+    }
+
     private String getNotFoundMessage(Long id) {
         return String.format("Book with id: %s not found", id);
     }
+
+    private Set<Category> categoriesIdToCategories(List<Long> categories) {
```

Comment:
can this method be delegated to mapper?

Reply:
Maybe, but I only need to set the categories that are in the database. That is, in any case, I have to make a request to the database. Therefore, there are three options, either leave it as it is, or use the repository in the mapper (but it seems to me that the mentors do not allow this)) ), or check through the "exist" (this will be a request to the database) + then use the mapper (it seems long )